src/gasPriceOracle/oracle.ts

```diff
@@ -1,51 +1,79 @@
 import assert from "assert";
-import { Transport } from "viem";
-import { providers } from "ethers";
+import { PublicClient, Transport } from "viem";
+import { PopulatedTransaction, providers } from "ethers";
 import { CHAIN_IDs } from "../constants";
 import { BigNumber, chainIsOPStack } from "../utils";
-import { GasPriceEstimate } from "./types";
+import { GasPriceEstimate, InternalGasPriceEstimate } from "./types";
 import { getPublicClient } from "./util";
 import * as arbitrum from "./adapters/arbitrum";
 import * as ethereum from "./adapters/ethereum";
 import * as linea from "./adapters/linea";
 import * as polygon from "./adapters/polygon";
-import * as arbitrumViem from "./adapters/arbitrum-viem";
 import * as lineaViem from "./adapters/linea-viem";
-import * as polygonViem from "./adapters/polygon-viem";
+import { PolygonGasStation } from "./adapters/polygon";
+
+export interface GasPriceEstimateOptions {
+  // baseFeeMultiplier Multiplier applied to base fee for EIP1559 gas prices (or total fee for legacy).
+  baseFeeMultiplier: number;
+  // legacyFallback In the case of an unrecognized chain, fall back to type 0 gas estimation.
+  legacyFallback: boolean;
+  // chainId The chain ID to query for gas prices. If omitted can be inferred by provider.
+  chainId: number;
+  // unsignedTx The unsigned transaction used for simulation by Linea's Viem provider to produce the priority gas fee.
+  unsignedTx?: PopulatedTransaction;
+  // transport Viem Transport object to use for querying gas fees used for testing.
+  transport?: Transport;
+  // polygonGasStation Custom Polygon GasStation class used for testing.
+  polygonGasStation?: PolygonGasStation;
+}
+
+const GAS_PRICE_ESTIMATE_DEFAULTS = {
+  baseFeeMultiplier: 1,
+  legacyFallback: true,
+};
 
 /**
  * Provide an estimate for the current gas price for a particular chain.
- * @param chainId The chain ID to query for gas prices.
  * @param provider A valid ethers provider.
- * @param legacyFallback In the case of an unrecognised chain, fall back to type 0 gas estimation.
- * @returns Am object of type GasPriceEstimate.
+ * @param {opts} GasPriceEstimateOptions optional parameters.
+ * @returns An  object of type GasPriceEstimate.
  */
 export async function getGasPriceEstimate(
   provider: providers.Provider,
-  chainId?: number,
-  transport?: Transport,
-  legacyFallback = true
+  opts: Partial<GasPriceEstimateOptions>
 ): Promise<GasPriceEstimate> {
-  chainId ?? ({ chainId } = await provider.getNetwork());
+  const baseFeeMultiplier = opts.baseFeeMultiplier ?? GAS_PRICE_ESTIMATE_DEFAULTS.baseFeeMultiplier;
+  assert(
+    baseFeeMultiplier >= 1.0 && baseFeeMultiplier <= 5,
+    `Require 1.0 < base fee multiplier (${baseFeeMultiplier}) <= 5.0 for a total gas multiplier within [+1.0, +5.0]`
+  );
+  const chainId = opts.chainId ?? (await provider.getNetwork()).chainId;
+  const optsWithDefaults: GasPriceEstimateOptions = {
+    ...GAS_PRICE_ESTIMATE_DEFAULTS,
+    ...opts,
```

Comment:
What's the rule in case of a conflict between `GAS_PRICE_ESTIMATE_DEFAULTS` and `opts` ? IS it the ordering that decides?

Reply:
Yeah that's how I would understand the syntax. So fill in defaults first, but allow user to override them. The test cases where I set the `baseFeeMultiplier = 2` seem to confirm this behavior

Reply:
This seems to test OK in the REPL at least:

```
> y = { 1: 1, 2: 2, 3: 3 }
{ '1': 1, '2': 2, '3': 3 }

> a = { 1: "a" }
{ '1': 'a' }

> b = { 1: "b" }
{ '1': 'b' }

> { ...y, ...a, ...b }
{ '1': 'b', '2': 2, '3': 3 }

> { ...y, ...b, ...a }
{ '1': 'a', '2': 2, '3': 3 }
 
> { ...y, ...b, ...a, 1: "x" }
{ '1': 'x', '2': 2, '3': 3 }
```